src/contracts/GPv2Settlement.sol

```diff
@@ -142,4 +158,58 @@ contract GPv2Settlement {
         );
         return IUniswapV2Pair(uint256(pairAddressBytes));
     }
+
+    /// @dev Verifies the specified clearing price is valid. Spefically, this
+    /// method verifies that the ±0.1% fee spread around the clearing price is
+    /// completely within the ±0.3% AMM fee spread around the Uniswap AMM price.
+    /// The AMM price is either defined by the token reserves (AKA spot price)
+    /// in the case the AMM is not traded with, or the actual swap amount (AKA
+    /// effective price) otherwise.
+    function verifyClearingPrice(
+        IUniswapV2Pair pair,
+        int112 d0,
+        int112 d1,
+        uint256 clearingPrice0,
+        uint256 clearingPrice1
+    ) internal view {
+        uint256 uniPrice0;
+        uint256 uniPrice1;
+        if (d0 | d1 == 0) {
+            // NOTE: In case the amounts swapped with the AMM is 0, we want to
+            // ensure that the clearing price with spread is within the AMM
+            // price spread.
+            (uint112 reserve0, uint112 reserve1, ) = pair.getReserves();
+            (uniPrice0, uniPrice1) = (uint256(reserve0), uint256(reserve1));
+        } else {
+            // NOTE: We want to check that exactly only one of `d0` or `d1` is
```

Comment:
Maybe a stupid idea but since I know you love bit-magic you could xor the MSB of d0 and d1 to check that?

Reply:
We're going to change this. See discussion in #107 on encoding the trade direction with the token ordering.